src/presentation/service/services.go

```diff
@@ -0,0 +1,470 @@
+	return &ServicesService{
+		persistentDbService: persistentDbService,
+	}
+}
+
+func (service *ServicesService) Read() ServiceOutput {
+	servicesQueryRepo := servicesInfra.NewServicesQueryRepo(service.persistentDbService)
+	servicesList, err := useCase.ReadServicesWithMetrics(servicesQueryRepo)
+	if err != nil {
+		return NewServiceOutput(InfraError, err.Error())
+	}
+
+	return NewServiceOutput(Success, servicesList)
+}
+
+func (service *ServicesService) ReadInstallables() ServiceOutput {
+	servicesQueryRepo := servicesInfra.NewServicesQueryRepo(service.persistentDbService)
+	servicesList, err := useCase.ReadInstallableServices(servicesQueryRepo)
+	if err != nil {
+		return NewServiceOutput(InfraError, err.Error())
+	}
+
+	return NewServiceOutput(Success, servicesList)
+}
+
+func (service *ServicesService) CreateInstallable(
+	input map[string]interface{},
+	shouldSchedule bool,
+) ServiceOutput {
+	requiredParams := []string{"name"}
+	err := serviceHelper.RequiredParamsInspector(input, requiredParams)
+	if err != nil {
+		return NewServiceOutput(UserError, err.Error())
+	}
+
+	name, err := valueObject.NewServiceName(input["name"])
+	if err != nil {
+		return NewServiceOutput(UserError, err.Error())
+	}
+
+	var versionPtr *valueObject.ServiceVersion
+	if input["version"] != nil {
+		version, err := valueObject.NewServiceVersion(input["version"])
+		if err != nil {
+			return NewServiceOutput(UserError, err.Error())
+		}
+		versionPtr = &version
+	}
+
+	var startupFilePtr *valueObject.UnixFilePath
+	if input["startupFile"] != nil {
+		startupFile, err := valueObject.NewUnixFilePath(input["startupFile"])
+		if err != nil {
+			return NewServiceOutput(UserError, err.Error())
+		}
+		startupFilePtr = &startupFile
+	}
+
+	envs := []valueObject.ServiceEnv{}
+	if input["envs"] != nil {
+		rawEnvs, assertOk := input["envs"].([]string)
+		if !assertOk {
+			return NewServiceOutput(UserError, "InvalidEnvs")
+		}
+
+		for _, rawEnv := range rawEnvs {
+			env, err := valueObject.NewServiceEnv(rawEnv)
+			if err != nil {
+				continue
+			}
+			envs = append(envs, env)
+		}
+	}
+
+	portBindings := []valueObject.PortBinding{}
+	if input["portBindings"] != nil {
+		rawPortBindings, assertOk := input["portBindings"].([]string)
+		if !assertOk {
+			return NewServiceOutput(UserError, "InvalidPortBindings")
+		}
+
+		for _, rawPortBinding := range rawPortBindings {
+			portBinding, err := valueObject.NewPortBinding(rawPortBinding)
+			if err != nil {
+				continue
+			}
+			portBindings = append(portBindings, portBinding)
+		}
+	}
+
+	var autoStartPtr *bool
+	if input["autoStart"] != nil {
+		var err error
+		autoStart, err := voHelper.InterfaceToBool(
+			input["autoStart"],
+		)
+		if err != nil {
+			return NewServiceOutput(UserError, "InvalidAutoStart")
+		}
+		autoStartPtr = &autoStart
+	}
+
+	var timeoutStartSecsPtr *uint
+	if input["timeoutStartSecs"] != nil {
+		var assertOk bool
+		timeoutStartSecs, assertOk := input["timeoutStartSecs"].(uint)
```

Comment:
No need to assert to uint if you can use InterfaceToUint. Here and on the next one.

Reply:
Done.